packages/stateManager/index.ts

```diff
@@ -1,104 +1,636 @@
+    super(store, client);
+  }
+  //Process all manifest events. Convert bytes to hex and save to shop store
+  async _processEvent(event: schema.ShopEvents): Promise<void> {
+    if (event.shopManifest) {
+      const sm = event.shopManifest;
+      const manifest = {
+        tokenId: bytesToHex(sm.shopTokenId),
+        name: sm.name,
+        profilePictureUrl: sm.profilePictureUrl,
+        publishedTagId: bytesToHex(sm.publishedTagId),
+        description: sm.description,
+        addAcceptedCurrencies: [],
+        setBaseCurrency: null,
+        addPayee: null,
+      };
+      await this.store.put("shopManifest", manifest);
+      this.emit("createShopManifest", manifest);
+      return;
+    } else if (event.updateShopManifest) {
+      const um = event.updateShopManifest;
+      const manifest = await this.store.get("shopManifest");
+
+      if (um.name) {
+        manifest.name = um.name;
+      }
+      if (um.description) {
+        manifest.description = um.description;
+      }
+      if (um.profilePictureUrl) {
+        manifest.profilePictureUrl = um.profilePictureUrl;
+      }
+      if (um.publishedTagId) {
+        manifest.publishedTagId = bytesToHex(um.publishedTagId);
+      }
+      if (um.setBaseCurrency) {
+        manifest.setBaseCurrency = {
+          chainId: Number(um.setBaseCurrency.toJSON().chainId),
+          tokenAddr: bytesToHex(um.setBaseCurrency.tokenAddr),
+        };
+      }
+      if (um.addAcceptedCurrencies) {
+        const currencies = [...manifest.addAcceptedCurrencies];
+        um.addAcceptedCurrencies.forEach((a: schema.IShopCurrency) => {
+          currencies.push({
+            tokenAddr: bytesToHex(a.tokenAddr),
+            chainId: Number(a.toJSON().chainId),
+          });
+        });
+        manifest.addAcceptedCurrencies = currencies;
+      }
+      if (um.removeAcceptedCurrencies) {
+        let filtered = [...manifest.addAcceptedCurrencies!];
+        for (const rm of um.removeAcceptedCurrencies) {
+          filtered = manifest.addAcceptedCurrencies!.filter(
+            (cur) => cur.tokenAddr !== bytesToHex(rm.tokenAddr),
+          );
+        }
+        manifest.addAcceptedCurrencies = filtered;
+      }
+      await this.store.put("shopManifest", manifest);
+      this.emit("updateShopManifest", manifest);
+
+      return;
+    }
+  }
+  async create(
+    manifest: CreateShopManifest,
+    shopId: `0x${string}`,
+  ): Promise<ShopManifest> {
+    //FIXME publishedTagId & profilePictureUrl are currently a required fields for ShopManifest
+    await this.client.shopManifest(
+      {
+        ...manifest,
+        publishedTagId: new Uint8Array(32),
+        profilePictureUrl: "https://http.cat/images/200.jpg",
+      },
+      shopId,
+    );
+    return new Promise((resolve, reject) => {
+      const remove = () => this.removeListener("createShopManifest", onUpdate);
+      function onUpdate(create: ShopManifest) {
+        if (create.tokenId === shopId) {
```

Comment:
same as above.

you need to use the eventId that you get from the network client and also propagate that back through the emitter.